backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -208,10 +208,24 @@ def sanitize_records(self, records, selected_trees, filters):
             status = r["status"]
             table_test = "boot" if is_boot(r) else "test"
 
-            test_filter_is_empty = len(filters_map[table_test]) == 0
+            test_filter_is_empty = len(filters_map['build']) == 0 and len(filters_map[table_test]) == 0
             test_pass_in_filters = self.pass_in_filters(r, filters_map[table_test])
 
-            jump_test = not test_filter_is_empty and not test_pass_in_filters
+            build_pass_in_filters = True
+
+            if build_id not in processed_builds:
+                processed_builds.add(build_id)
+                build = get_build(r, current_tree["index"])
+
+                if len(filters_map['build']) == 0 or self.pass_in_filters(build, filters_map['build']):
+                    filtered_builds.add(build_id)
```

Comment:
You are adding builds to this `filtered_builds` set but not using it anywhere else